rotkehlchen/history/events.py

```diff
@@ -507,15 +514,18 @@ def populate_history_cb(
             asset_movements.extend(result_asset_movements)
 
             if exchange_specific_data:
-                # This can only be poloniex at the moment
-                polo_loans_data = exchange_specific_data
-                loans.extend(process_polo_loans(
-                    msg_aggregator=self.msg_aggregator,
-                    data=polo_loans_data,
-                    # We need to have history of loans since before the range
-                    start_ts=Timestamp(0),
-                    end_ts=end_ts,
-                ))
+                if all(isinstance(i, LedgerAction) for i in exchange_specific_data):
```

Comment:
I see what you are doing here. But this won't scale. We did not need to process exchange specific data per exchange before.

I think you would need to edit the `ExchangeHistorySuccessCallback` and add one more argument as the first argument. And that is the location of the exchange (thus passing the type of exchange to the callback).

And then do this if depending on that type. Does this make sense?

Reply:
Sure I considered adding a location argument when I discovered Poloniex was the only exchange using this callback.  It should only run the code if the callback returned a value for exchange_specific_data.